app/utilities/AdminAuthenticator.java

```diff
@@ -0,0 +1,35 @@
+package utilities;
+
+import models.Token;
+import models.User;
+import play.Play;
+import play.mvc.Http;
+
+/**
+ * Created by Ejub on 02/02/16.
+ * Class AdminAuthenticator can be used to protect specific routes by checking if administrator is authenticated.
+ */
+public class AdminAuthenticator extends UserAuthenticator{
+    /**
+     * Retrieves the user out of the http context.
+     * Method firstly checks if authentication token found in context is valid, and whether it matches to a specified administrator.
+     *
+     * @param ctx the http context
+     * @return the user is returned if token is valid and if user is confirmed
+     */
+    @Override
+    public String getUsername(Http.Context ctx) {
+        String token = getTokenFromHeader(ctx);
+        if (token != null) {
+            Token tempToken = PersistenceManager.getToken(token);
+            if (tempToken == null) {
+                return null;
+            }
+            User user = PersistenceManager.getUserByEmail(tempToken.getEmail());
+            if (user != null) {
+                return user.getEmail().equals(Play.application().configuration().getString("admin_email")) ? user.getEmail() : null;
+            }
+        }
+        return null;
```

Comment:
Still there is repeating code. This is what you should do:

- call super.getUsername(ctx)
- in UserAuthenticator put loaded user inside of HTTP Context like this: ctx.args.put(@"CurrentUser", user) in case user is authenticated
- then load user from context here and check user's role

Also, I'm not sure what is admin_email? This is wrong and should not be done like this. There should be admin flag in DB, or list of roles associated with user.